server/src/main/java/org/apache/druid/segment/realtime/appenderator/UnifiedIndexerAppenderatorsManager.java

```diff
@@ -248,6 +289,50 @@ private AppenderatorConfig rewriteAppenderatorConfigMemoryLimits(AppenderatorCon
     return new MemoryParameterOverridingAppenderatorConfig(baseConfig, perWorkerLimit);
   }
 
+  private class DatasourceBundle
+  {
+    private final SinkQuerySegmentWalker walker;
+    private final Map<String, List<Appenderator>> taskAppenderatorMap;
```

Comment:
I'm not sure if there's any task type which uses two or more appenderators. Would you point me out what task is doing it?

Reply:
The CompactionTask creates multiple IndexTask instances and runs them in sequence (each will create its own appenderator). I updated the CompactionTask to use the same task ID for the index tasks it runs internally.

Reply:
Ah yeah, I forgot about the compactionTask. Thanks.